### DISCUSSION TITLE
What is the data basis for drawing graphs in the SERVICE GRAPH module

### DISCUSSION THREAD

**Discussion by QuanZi123** [uptrace/uptrace#357] [Q&A] [answered]
I try to use lua script to report trace data of opentelemetry, and the access trace can be correctly displayed in <view trace>, but it fails to be displayed in <service graph> module. I want to know what uptrace draws the graph according to. And figure out what's missing from my lua script

**Comment by QuanZi123** [comment]
@vmihailenco

**Comment by vmihailenco** [comment, marked as answer]
### General attributes

- `service.name` is an attribute that represents the logical name of the service. Required.

- `_kind` (`span_kind`) categorizes the type of work being represented by the span. Make sure your traces have `client/server` or `producer/consumer` span pairs. Required.

- `_status_code` (`span_status_code`) provides information about the result of a specific operation or unit of work represented by this span. This attribute helps to monitor and understand the success, failure, or other conditions of a given operation. Recommended.

### RPC calls

- `rpc.service` represents the full (logical) name of the service being called. Optional.

### DB calls

- `db.name` contains the name of the database being accessed. Recommended if you have multiple databases.

### HTTP calls

- `server.socket.domain` represents the domain name of an immediate peer. Required for HTTP calls.

- `server.socket.address` represents the physical server IP address or Unix socket address. Required for HTTP calls if `server.socket.domain` is not provided.

### Messaging calls

- `messaging.destination.name` is the message destination name, for example, `MyQueue`, `MyTopic`.

- `messaging.client_id` is a unique identifier for the client that consumes or produces a message.

- `messaging.kafka.consumer.group` is the name of the Kafka Consumer Group that is handling the message. Only applies to consumers, not producers.
